lib/req/request.ex

```diff
@@ -260,6 +260,43 @@ defmodule Req.Request do
     update_in(request.error_steps, &(steps ++ &1))
   end
 
+  @doc """
+  Runs the request steps in the request pipeline but does not execute them.
+
+  Accepts a `Req.Request` struct, or the same keyword list of options accepted
+  by `Req.new/1`.
+
+  Returns a `Req.Request` struct with all of its steps executed.
+
+  ## Examples
+
+      iex> prepared = Req.Request.prepare(method: :get, base_url: "http://www.example.com", params: [x: 1])
+      iex> prepared.url.host
+      "www.example.com"
+      iex> prepared.url.query
+      "x=1"
+  """
+  def prepare(options) when is_list(options) do
+    options
+    |> Req.new()
+    |> prepare()
+  end
+
+  def prepare(%Req.Request{request_steps: []} = request) do
+    request
+  end
+
+  def prepare(%{request_steps: [step | steps]} = request) do
+    case run_step(step, request) do
+      %Req.Request{} = request ->
+        request = %{request | request_steps: steps}
+        prepare(request)
+
+      {_request, %{__exception__: true} = exception} ->
```

Comment:
please add a test for this clause. Also, I'm not sure if we should raise the exception we got, I feel like we should raise an exception saying that we cannot prepare a request with a step that returns an exception and then print the exception. Similarly, if a request step returns {request, response}, we should raise too.